ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantSession.scala

```diff
@@ -22,29 +27,61 @@ private[participant] final class ParticipantSession(
 
   private[this] val logger = LoggerFactory.getLogger(classOf[ParticipantSession])
 
+  private[this] val eventually = RetryStrategy.exponentialBackoff(10, 10.millis)
+
   private[this] val services: LedgerServices = new LedgerServices(channel)
 
-  private[this] val eventually = RetryStrategy.exponentialBackoff(10, 10.millis)
+  // The ledger identifier is retrieved only once when the participant session is created
+  // Changing the ledger identifier during the execution of a session can result in unexpected consequences
+  // The test tool is designed to run tests in an isolated environment but changing the
+  // global state of the ledger breaks this assumption, no matter what
+  private[this] val ledgerIdF =
+    eventually { attempt =>
+      logger.debug(s"Retrieving ledger identifier to create ledger context (attempt #$attempt)...")
+      services.identity.getLedgerIdentity(new GetLedgerIdentityRequest).map(_.ledgerId)
+    }
+
+  // The time-to-live for commands defaults to the maximum value as defined by the ledger
+  // configuration and can be adjusted down. Regardless the output of the adjustment, the
+  // value is always going to be clipped by the minimum and maximum configured values.
+  // Note that this value is going to never change after it's read the first time, so changing
+  // ledger configuration across tests may have wildly unexpected consequences. In general the
+  // test tool is designed to have tests work in isolation and tests addressing changes to the
+  // global state of the ledger should be isolated in their own test runs.
+  private[this] val ttlNanosF: Future[Long] =
+    ledgerIdF
+      .flatMap { id =>
+        SingleItemObserver
+          .first[GetLedgerConfigurationResponse](services.configuration
+            .getLedgerConfiguration(new GetLedgerConfigurationRequest(id), _))
+          .map(_.get.getLedgerConfiguration)
+      }
+      .map { configuration =>
+        val factor = config.commandTtlFactor
+        val min = (configuration.getMinTtl.seconds * 1E9) + configuration.getMinTtl.nanos.toDouble
```

Comment:
`1E9` <- what a rare sight to behold :)

Reply:
Alternatively:
```
import scala.concurrent.duration._

val min = (configuration.getMinTtl.seconds.seconds + configuration.getMinTtl.nanos.nanos
// ...
// multiplying by factor and the min max functions would work just the same
```

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3052/commits/f28736e5df07b0d30fd1b1d174e63fa974c75fde